### PR TITLE
Add Docker image artifacts

### LINKED ISSUE #290: Add docker image artifact to CI
## Background
Oftentimes, it is nice to be able to have access to a docker image that is generated from the PR, to test code locally, in case the launch test is unable to test a new piece of code. It takes a long time to build the image locally, as compared to building on github, especially on non-linux hosts.

## Task
Generate a tar.gz file of the image (already being done for core, but do it for onboard and landside) and save them as [artifacts](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts) (only if requested, perhaps via a comment on the PR). Make sure it's not deleted, one of the current steps in the pipeline deletes all artifacts generated. Also create a workflow that deletes all artifacts on a nightly basis, so that we don't forget to delete them after we download them.

### PR DESCRIPTION
**Pull request by ShaanGondalia**
Closes #290 

Creates onboard/landside artifacts if the "artifact" tag is present on a pull request. Also adds a cleanup script that runs once a day to remove all extra artifacts. Will test functionality on this pr.

### CODE REVIEW

**Comment by ShaanGondalia** [comment]
Tested build with and without "artifact" label, artifacts are created as expected. Artifact tag adds ~7 minutes to total build time, bottlenecked by arm64 onboard build workflow (Total time increases from ~35min to ~41min) . Ready to merge after a review.

**Review by muthuArivoli — COMMENTED** [review]
Two questions, see below, but otherwise looks good! Thanks for working on this stuff!

File: .github/workflows/build.yml
Comment (ShaanGondalia) [reply .github/workflows/build.yml:10]: I'll disable the builds if the event is a labeling, overlooked that

File: .github/workflows/build.yml
Comment (ShaanGondalia) [reply .github/workflows/build.yml:108]: This was by design. My goal was for it to be a kind of switch, so that artifacts wont be created on every build, just certain ones. The other solution I was looking into was triggering the event on a comment, which is probably a better choice for this. I'll look into that

**Comment by ShaanGondalia** [comment]
After implementing some changes, the new system triggers a build and creates docker image artifacts any time a user comments `/build-artifacts` on a PR. The existing functionality of triggering builds on PR updates and push events has not been changed.

**Comment by muthuArivoli** [comment]
Have you been able to validate this? The issue_comment action won't actually trigger unless the file is in master, which means to validate you'll have to create a test repo (or a fork) with these files in master and try to check using that. I just tried testing this in my own test repo, and it looks like its failing: https://github.com/muthuArivoli/robosub-test/runs/8182081649?check_suite_focus=true. Lmk if you just want to use that repo for testing, and I can add you.

**Comment by ShaanGondalia** [comment]
OK, tested pretty much every use case here: https://github.com/ShaanGondalia/robosub-ros/pull/2. I opted to move the comment trigger to a different workflow. I wanted to avoid this at first because a lot of the code is repeated, but I think the steps now are different enough to warrant them being separate workflows. I wanted to avoid having a bunch of conditionals in each step to have different behavior if the event was a pr or pr comment (the main issue here being checking out the correct code). I believe the delete-artifacts workflow worked correctly in your test repo. Let me know if there are any other cases you'd like to test before we merge.

**Review by muthuArivoli — APPROVED** [review]
Sounds good, and LGTM!

**Pull request merged**
